android/modules/ui/src/java/ti/modules/titanium/ui/widget/searchview/TiUISearchView.java

```diff
@@ -177,4 +163,98 @@ public void release()
+		}
+	}
+
+	private void updateIconSearchColor(int colorId)
+	{
+		try {
+			// Get the Search Button ImageView from the layout.
+			int id = TiRHelper.getResource("id.search_button");
+			ImageView buttonIV = searchView.findViewById(id);
+			// Apply a color filter with the proxy's color property.
+			buttonIV.setColorFilter(colorId);
+			// When iconified is disabled the view uses an imageView with
+			// the search_mag_icon id, so we will apply the color filter on it too.
+			// In iconified mode this view is not visible.
+			int idMagIcon = TiRHelper.getResource("id.search_mag_icon");
+			ImageView magIconImageView = searchView.findViewById(idMagIcon);
+			magIconImageView.setColorFilter(colorId);
+		} catch (ResourceNotFoundException e) {
+			Log.e(TAG, "Could not find SearchView Search Button");
+		}
+	}
+
+	private void updateIconSearchCloseColor(int colorId)
+	{
+		try {
+			// Get the Close Search Button ImageView from the layout.
+			int closeButtonId = TiRHelper.getResource("id.search_close_btn");
+			ImageView buttonIV = searchView.findViewById(closeButtonId);
+			// Apply a color filter with the proxy's color property.
+			buttonIV.setColorFilter(colorId);
+		} catch (ResourceNotFoundException e) {
+			Log.e(TAG, "Could not find SearchView Search Close Button");
+		}
+	}
+
+	private void updateIconSearchHintColor(int colorInt)
+	{
+		try {
+			// Get the EditText instance from the layout.
+			EditText editText = searchView.findViewById(TiRHelper.getResource("id.search_src_text"));
+			// Set the default textSize from Appcompat
+			int textSize = (int) ((double) editText.getTextSize() * 1.25D);
+			// Get the Search icon drawable.
+			Drawable searchIconDrawable;
+			searchIconDrawable = getProxy().getActivity().getResources().getDrawable(
+				TiRHelper.getResource("drawable.abc_ic_search_api_material"), null);
+			// Make it mutable to allow different states of it to be used.
+			searchIconDrawable.mutate();
+			// Apply a color filter with the color from the proxy's property.
+			searchIconDrawable.setColorFilter(colorInt, PorterDuff.Mode.SRC_IN);
+			// Set the default bounds from Appcompat
+			searchIconDrawable.setBounds(0, 0, textSize, textSize);
+			// Create the SpannableStringBuilder.
+			SpannableStringBuilder ssb = new SpannableStringBuilder("   ");
+			// Set the new ImageSpan.
+			ssb.setSpan(new ImageSpan(searchIconDrawable), 1, 2, 33);
+			ssb.append("");
+			// In order to stick to the default behavior we need to manually add the hint text,
+			// if there is one, to the newly created SpannableString. This is necessary because
+			// calling setQuery method of the SearchView triggers the default getDecoratedHint
+			// method that will create a new ImageSpan without the custom color filter.
+			if (getProxy().hasPropertyAndNotNull(TiC.PROPERTY_HINT_TEXT)) {
+				ssb.append(getProxy().getProperty(TiC.PROPERTY_HINT_TEXT).toString());
+			}
+			// Add it as a hint.
+			editText.setHint(ssb);
```

Comment:
^ The above makes me nervous. It makes assumptions about the internal implementation details on Google's end, which could change in the future. Is there really no other nice way to do this? I was able to replace this icon via a theme/style, except when the `SearchView` was embedded within the `ActionBar`.

Reply:
I mean, you might be right. There might be no other way to do this without hacks... but we then have to be concerned about potential breakage/issues in the future.

Reply:
From what I tested providing a custom theme with overridden properties for close, hintIcon and searchIcon worked just fine. The case where I embedded the SearchView as a Menu item in the ActionBar required to separately pass the drawable I used for `<item name="searchIcon">@drawable/pictureduo</item>` because it is different from the already embedded resources which are referenced by ID (like the sample `icon: Ti.Android.R.drawable.ic_menu_search,`)
Possibly the biggest benefit would be that we can have partial parity with the properties of Ti.UI.SearchBar, but that leaves us at "Is it worth it to have that at the price of blindly assuming what is happening under the AppCompat library?" Maybe we can discuss this in details offline ?